### ISSUE TITLE
Iteration Plan for March 2021

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode#118334]
This plan captures our work in **March**. This is a 4-week iteration. We will ship ~~in **early April**~~ **end of March** :smiley:.

## Endgame
- **March 22, 2021**: Endgame begins
- **March 26, 2021**: Endgame done

The endgame details for this iteration are tracked [here](https://github.com/microsoft/vscode/issues?q=is%3Aissue+label%3Aendgame-plan+milestone%3A%22March+2021%22).

## Plan Items
Below is a summary of the top level plan items.

Legend of annotations:

| Mark                 | Description                                       |
| -------------------- | ------------------------------------------------- |
| :runner:             | work in progress                                  |
| :hand:               | blocked task                                      |
| :muscle:             | stretch goal for this iteration                   |
| :red_circle:         | missing issue reference                           |
| :large_blue_circle:  | more investigation required to remove uncertainty |
| :black_circle:       | under discussion within the team                  |
| :black_large_square: | a large work item, larger than one iteration      |

### Accessibility
- [x] Accessibility issues, see [query](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility+milestone%3A%22March+2021%22) @isidorn **team**
- [ ] :runner: Assess notebook accessibility https://github.com/microsoft/vscode/issues/111255 @rebornix @isidorn @roblourens
- [x] Investigate in accessibility support for multiple cursors https://github.com/microsoft/vscode/issues/109918 @isidorn

### Workbench
- [ ] :runner: Trusted Workspaces https://github.com/microsoft/vscode/issues/106488 @lszomoru @sbatten
  - [ ] :runner: Finalize API
  - [ ] :runner: Provide documentation for extension authors on adoption and testing
  - [ ] :runner: Begin engagement with extension authors
- [x] Show tab decorations by default https://github.com/microsoft/vscode/issues/118103 @jrieken
- [x] Provide "Copy As..." as a submenu https://github.com/microsoft/vscode/issues/118208 @joaomoreno
- [ ] :runner: Improve the welcome pages https://github.com/microsoft/vscode/issues/106715 @jacksonkearl @misolori @digitarald
  - [x] Explore consolidation of welcome and getting started @misolori @jacksonkearl @digitarald
  - [ ] :muscle: Explore integration of release notes and extension changelogs @jacksonkearl
  - [ ] :runner: Continue user testing on new welcome pages @digitarald
  - [ ] :runner: More context for primary section https://github.com/microsoft/vscode/issues/116976 @digitarald
  - [ ] :runner: Update content: Resized images for dark and light theme https://github.com/microsoft/vscode/issues/115325 https://github.com/microsoft/vscode/issues/115327 @digitarald
  - [ ] :runner: Add content for experienced users https://github.com/microsoft/vscode/issues/118281 @digitarald
  - [ ] :runner: Explore Extension content contributions https://github.com/microsoft/vscode/issues/118402 @digitarald
- [x] Improve handling of untitled files: investigate into automatic language classification https://github.com/microsoft/vscode/issues/118455  @isidorn

### UX
- [x] Update app icon for macOS Big Sur https://github.com/microsoft/vscode/issues/101014 @misolori
- [ ] :muscle: Continue exploration of profiles https://github.com/microsoft/vscode/issues/116740 @misolori
- [ ] :muscle: Sketch out a classification for notifications and usage guidance for each https://github.com/microsoft/vscode/issues/119463 @misolori @isidorn

### Monaco Editor
- [x] Publish a new version of the Monaco editor @alexdima

### Notebook Editor
- [ ] :runner: Review Notebook API https://github.com/microsoft/vscode/issues/93265 @jrieken @rebornix @roblourens
- [ ] :runner: Enable extensible markdown rendering for notebooks https://github.com/microsoft/vscode/issues/106701 @mjbvz @rebornix
- [ ] :runner: Reduce notebook papercut issues https://github.com/microsoft/vscode/issues/113245 @rebornix @jrieken @roblourens
- [x] Allow cell folding on double-click https://github.com/microsoft/vscode/issues/113248 @roblourens
- [x] Support cell multi-select https://github.com/microsoft/vscode/issues/114156 @rebornix
- [x] Support filters in the notebook diff editor @rebornix https://github.com/microsoft/vscode/issues/118933
- [x] Clarify relationship between notebook and workspace trust https://github.com/microsoft/vscode/issues/118584 @roblourens
- [x] Improve the `List` widget to better support notebook use cases https://github.com/microsoft/vscode/issues/118044 https://github.com/Microsoft/vscode/issues/118045 @joaomoreno

### Tasks
- [x] Improve the '"applyTo": "closedDocuments' support in the problem matcher https://github.com/microsoft/vscode/issues/116760 @alexr00
- [x] Do not show the NPM Scripts explorer when the project doesn't have a `package.json` https://github.com/microsoft/vscode/issues/117950 @alexr00

#### CSS
- [x] Investigate into an improved CSS grammar https://github.com/microsoft/vscode/issues/115480 @alexr00

#### LSP /LSIF
- [ ] :runner: LSP - Add support for a [pull model for diagnostics](https://github.com/microsoft/language-server-protocol/issues/737) @dbaeumer
- [ ] :runner: LSIF - Add support for [search scopes](https://github.com/microsoft/lsif-node/issues/127) @dbaeumer

#### JavaScript/TypeScript
- [x] Adopt TS 4.2.x recovery build @mjbvz
- [x] Validate the new semantic highlighter for TS https://github.com/microsoft/vscode/issues/119139 @aeschli

### Debug
- [x] Inline break point editing should support to edit both the hit count and the expression https://github.com/microsoft/vscode/issues/115111 @isidorn
- [x] Surface DataBreakpoint access types in the UI https://github.com/microsoft/vscode/issues/113922 @isidorn @weinand
- [ ] :hand: Ability to drag and drop breakpoints https://github.com/microsoft/vscode/issues/88227 @isidorn
- [x] Clear terminal before launching another debuggee in terminal https://github.com/microsoft/vscode/issues/116699 @weinand
- [ ] :runner: Improve browser tab notifications for debugging in Web UI https://github.com/microsoft/vscode/issues/100757 @weinand @isidorn
- [ ] :muscle: Investigate debug UI for highly concurrent languages https://github.com/microsoft/vscode/issues/116109 @weinand @connor4312

#### DAP
- [x] Support to detect which breakpoint caused a Stopped event https://github.com/microsoft/debug-adapter-protocol/issues/178 @weinand
- [ ] :runner: Provide more control over the disconnect from target behavior https://github.com/microsoft/debug-adapter-protocol/issues/175 https://github.com/microsoft/debug-adapter-protocol/issues/177 @weinand

### Testing
- [ ] :runner: Explore how to improve testing support https://github.com/microsoft/vscode/issues/107467 @connor4312 @sandy081

### Terminal
- [x] Address issues with <kbd>Ctrl+C</kbd> on iPadOS https://github.com/microsoft/vscode/issues/114009 @meganrogge @tyriar
- [x] Make the WebGL renderer the default, improve fallback logic https://github.com/microsoft/vscode/issues/118064 @meganrogge @tyriar
- [x] Polish terminal restart caused by changes of environment variable collection(s) https://github.com/microsoft/vscode/issues/118066 @tyriar @meganrogge
- [x] Explore replacing the terminal dropdown with tabs https://github.com/microsoft/vscode/issues/10546 @tyriar @meganrogge
- [x] Terminal profiles https://github.com/microsoft/vscode/issues/45444 @meganrogge @tyriar

### Source Control
- [ ] :muscle: Explore UI additions to help guide users to push to a remote after committing https://github.com/microsoft/vscode/issues/110882 @eamodio
- [ ] :muscle: Investigate fine grained change events in git core for consumption in the GHPRI and GitLens extensions https://github.com/microsoft/vscode/issues/116360 @eamodio

### API
- API finalization: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22March+2021%22+label%3A%22api-finalization%22)
- API proposals: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22March+2021%22+label%3A%22api-proposal%22)

### Extensions
- [x] Explore the concept of "extension plug-in" https://github.com/microsoft/vscode/issues/118442 @sandy081 @misolori
- [x] Detect default branch in `vsce` https://github.com/microsoft/vscode-vsce/issues/534 @joaomoreno
- [x] Add a `Marketplace` top level category to the Issue Reporter https://github.com/microsoft/vscode/issues/118473 @rmacfarlane

### Extension Contributions
- [x] Improvements to the Remote extensions https://github.com/microsoft/vscode-remote-release/issues/4609 **team**
- [x] Improvements to the GitHub PR extension @rmacfarlane @eamodio
  - [x] Enable PR focus mode by default https://github.com/microsoft/vscode-pull-request-github/issues/2559 @rmacfarlane
  - [ ] :runner: Improve refresh behavior of the PR views https://github.com/microsoft/vscode-pull-request-github/issues/2478 @rmacfarlane
  - [x] Explore automatically setting upstream when encountering forks https://github.com/microsoft/vscode-pull-request-github/issues/2439 @rmacfarlane
  - [ ] :muscle: Explore support for resolving comments https://github.com/microsoft/vscode-pull-request-github/issues/339 @rmacfarlane
  - [x] Contribute Copy Permalink to the 'Copy As' submenu https://github.com/microsoft/vscode-pull-request-github/issues/2326 @alexr00

### Engineering
- [ ] :runner: Consolidate `Open with` across text editors, custom editors and notebooks, see https://github.com/microsoft/vscode/issues/100281 @lramos15
- [ ] :runner: Explore consolidation of saving across text editors, custom editors and notebooks https://github.com/microsoft/vscode/issues/117873 @bpasero
- [x] Consolidate local and remote terminal implementations https://github.com/microsoft/vscode/issues/116467 @Tyriar
- [ ] :runner: Consolidate publishing of build artifacts https://github.com/microsoft/vscode/issues/117891 @TylerLeonhardt @joaomoreno
- [x] Fix crash in vscode-encrypt when encryption service fails on POSIX platforms https://github.com/microsoft/vscode/issues/116690
- [ ] :runner: Scroll performance on Windows https://github.com/microsoft/vscode/issues/107016 @deepak1556
- [ ] :runner: Support scoped Find on Page in Chromium to unblock our transition from `webview` to `iframe` for custom editors and views @rzhao271 https://github.com/microsoft/vscode/issues/96307
- [ ] :muscle: :red_circle: Adopt `vscode.open` and `vscode.diff` in the GHPRI extension @alexr00
- [ ] :runner: Assume ownership of our quick picker code https://github.com/microsoft/vscode/issues/118586 @TylerLeonhardt
- [x] Introduce an `IProcessEnvService` for resolving the merged process environment based on context https://github.com/microsoft/vscode/issues/108804 @bpasero
- [x] Adopt new table widget in the Keyboard Shortcuts editor https://github.com/microsoft/vscode/issues/118123 @sandy081
- [x] Speed up mac builds https://github.com/microsoft/vscode/pull/116984 @TylerLeonhardt @deepak1556 @joaomoreno

#### Node.js-free, sand-boxed renderer
- [ ] :runner: Backlog https://github.com/microsoft/vscode/issues/92164 @bpasero @deepak1556

------
### Deferred
- LSP - Collaborate on "TypeHierarchy" support with the Java extension team @dbaeumer **Java team**
- Investigation: Cache VS Code extension builds https://github.com/microsoft/vscode/issues/115023 @joaomoreno @TylerLeonhardt
- Settings sync - debounce requests when a request is in progress https://github.com/microsoft/vscode/issues/91286 @sandy081
- Continue improving and adding functionality to REST Book https://github.com/tanhakabir/rest-book/issues/30 @tanhakabir
- Provide a way to override the editor input as part of `createEditorInput` https://github.com/microsoft/vscode/issues/100281 @lramos15 @bpasero

**Comment by sunliangqin** [comment]
Any plan to update the roadmap for 2021? https://github.com/microsoft/vscode/wiki/Roadmap

**Comment by kieferrm** [comment]
@sunliangqin yes, it's coming.

**Comment by TylerLeonhardt** [comment]
Endgame plan here: https://github.com/microsoft/vscode/issues/119494

**Comment by mgsergio** [comment]
You're awesome, guys! Thank you all for such a great tool!

**Comment by Frinando** [comment]
Hello world

**Comment by Frinando** [comment]
Hello world
